File: app/src/main/java/com/cookplan/recipe/import_recipe/parser/ParserFactory.java
package com.cookplan.recipe.import_recipe.parser;

/**
 * Created by DariaEfimova on 09.06.17.
 */

public class ParserFactory {

    public static final Parser createParser(String url) {
        if (url.contains("eda.ru")) {
            return new EdaHtmlParser(url);
        }
        if (url.contains("www.edimdoma.ru")) {
            return new EdimDomaHtmlParser(url);
        }
        return null;
    }
}

File: app/src/main/java/com/cookplan/recipe/import_recipe/parser/ParserFactory.kt
package com.cookplan.recipe.import_recipe.parser

import android.content.Context

/**
 * Created by DariaEfimova on 09.06.17.
 */

object ParserFactory {

    fun createParser(url: String, getContext: () -> Context?): Parser? {
        if (url.contains("eda.ru")) {
            return EdaTextParser(url, getContext)
        }
        return if (url.contains("www.edimdoma.ru")) {
            EdimDomaHtmlParser(url, getContext)
        } else null
    }
}